src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -2516,9 +2516,17 @@ components:
       pattern: ^[a-zA-Z0-9][_a-zA-Z0-9]*$
       type: string
       description: >
-        Dataset, table, and column names follow this pattern. This should be used for the name of any object in the
+        Table and column names follow this pattern. This should be used for the name of any object in the
         system. It enforces BigQuery naming rules except it disallows a leading underscore so we avoid collisions with
-        any extra columns the DR adds. For table names, this is shorter than what BigQuery allows.
+        any extra columns the DR adds. For table and column names, this is shorter than what BigQuery allows.
+    DatasetSnapshotNameProperty:
+      maxLength: 511
```

Comment:
Not sure what the conventions are for this, but you can also do this in the pattern regex.
```
      pattern: ^[a-zA-Z0-9][_a-zA-Z0-9]{1,511}$
```

Reply:
Nice! I just went with `maxLength` here to be consistent with the other validation properties